client/state/notices/types.ts

```diff
@@ -30,6 +30,7 @@ export interface BaseNoticeOptions {
 	isPersistent?: boolean;
 	onClick?: () => void;
 	showDismiss?: boolean;
+	onDismissClick?: () => void;
```

Comment:
Looking at [code](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/global-notices/index.jsx#L25), onDismissClick was supposed to be here 🤷‍♂️